lib/session.rb

```diff
@@ -0,0 +1,27 @@
+# typed: true
+# frozen_string_literal: true
+
+module AutoHCK
+  class Session < T::Struct
+    extend T::Sig
+    extend Models::JsonHelper
+
+    prop :cli, CLI
+
+    sig { params(workspace_path: String, cli: CLI, logger: AutoHCK::MultiLogger).void }
+    def self.save_cli(workspace_path, cli, logger)
+      File.write("#{workspace_path}/session.json", cli.serialize.to_json)
+      logger.info("Session saved to #{workspace_path}/session.json")
```

Comment:
It's not saving `Session` but saving `CLI`.

Reply:
Indeed

Reply:
```
    def self.save(workspace_path, logger)
      File.write("#{workspace_path}/session.json", self.to_json)
```